ssl/record/record.h

```diff
@@ -122,6 +122,8 @@ typedef struct record_layer_st {
     /* Record layer data to be processed */
     TLS_RECORD tlsrecs[SSL_MAX_PIPELINES];
 
+    /* DTLS epoch zero write sequence number */
+    uint64_t wlayer_epoch_zero_sequence;
```

Comment:
From the PR I only spot this variable set/get in `ssl_set_new_record_layer`. Could we limit the scope to that function?

Reply:
I don't think so. I need to save it outside of the record layer. What is happening with the Early data Hello Retry request test is we go from Epoch 0 then to Epoch 1 and back to Epoch 0. Each Epoch is a new record layer. When the server does this and gets the Client Hello the Server hasn't done the same dance as the client. Instead it wants the sequence number to be 7 (in case of the specific test) instead the client would send it as zero since it created a new record layer for the second Epoch 0. In reality it should be 7. Thus we need to store this outside of the record layer. Though if you have another suggestion on how to better do this I'm all ears.